src/webots/gui/WbView3D.cpp

```diff
@@ -1943,9 +1943,21 @@ void WbView3D::mouseMoveEvent(QMouseEvent *event) {
     if (mDisabledUserInteractionsMap.value(WbAction::DISABLE_FORCE_AND_TORQUE, false))
       // user interaction disabled
       return;
-    WbSolid *const selectedSolid = dynamic_cast<WbSolid *>(mPickedMatter);
-    if (!selectedSolid || selectedSolid->bodyMerger() == NULL)
+
+    WbNode *node = dynamic_cast<WbNode *>(mPickedMatter);
+    WbSolid *selectedSolid = NULL;
+    while (node) {
+      selectedSolid = dynamic_cast<WbSolid *>(node);
+      if (selectedSolid && selectedSolid->bodyMerger() != NULL)
+        break;
+
+      node = node->parentNode();
+      if (!node || node->level() < 1)  // abort the search at the top of this node chain
+        return;
+    }
+    if (!selectedSolid)
       return;
```

Comment:
I believe the logic is correct, however, the code could be slightly simplified:

```suggestion
    if (!node)
      return;
    WbSolid *selectedSolid;
    while (1) {
      selectedSolid = dynamic_cast<WbSolid *>(node);
      if (selectedSolid && selectedSolid->bodyMerger() != NULL)
        break;

      node = node->parentNode();
      if (!node || node->level() < 1)  // abort the search at the top of this node chain
        return;
    }
```

Reply:
Indeed, thank you.